pkgs/development/python-modules/duckdb/default.nix

```diff
@@ -35,21 +36,24 @@ buildPythonPackage rec {
       substituteInPlace setup.py \
         --replace-fail "ParallelCompile()" 'ParallelCompile("NIX_BUILD_CORES")' \
         --replace-fail "define_macros.extend([('DUCKDB_EXTENSION_AUTOLOAD_DEFAULT', '1'), ('DUCKDB_EXTENSION_AUTOINSTALL_DEFAULT', '1')])" "pass"
+
+      substituteInPlace pyproject.toml \
+        --replace-fail 'setuptools_scm>=6.4,<8.0' 'setuptools_scm'
```

Comment:
This would also make a lovely upstream patch.

Reply:
Hello, my I ask a few questions for my own understanding and learning? 

Was this change necessary due to failures on a specific architecture?

When I attempted this update, I only encountered problems with `test/issues/general/test_17757.test`, and there was no need to modify other parts of the derivation. At least on aarch64 linux and macOS.

Additionally, there is an excluded test, which is linked to an upstream issue that has now been closed, and the test passes. When should we take care of this type of cleanup? (I'm talking about `test/sql/copy/file_size_bytes.test`)